src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests.cs

```diff
@@ -64,6 +64,20 @@ public void RecordToRecordAggregateCoercionTest(string expr)
             Assert.IsNotType<ErrorValue>(result);
         }
 
+        [Theory]
+        
+        [InlineData("[{a:0},{a:\"3\",b:2}]", "Table({a:0,b:If(false,0)},{a:3,b:2})")]
+        [InlineData("[{a:1}, {a:true, b:2}]", "Table({a:1,b:If(false,0)},{a:1,b:2})")]
+        public void RecordToRecordAggregateCoercionDontDropFieldsTest(string expr, string expected)
+        {
+            var iSetup = InternalSetup.Parse($"TableSyntaxDoesntWrapRecords");
+            var engine = new RecalcEngine(new PowerFxConfig(Features.TableSyntaxDoesntWrapRecords));
+
+            var result = engine.Eval(expr) as TableValue;
```

Comment:
>as TableValue; [](http://example.com/codeflow?start=43&length=14)

nit:
(TableValue) engine.Eval(expr)

will give us an exception immediately at the cast. 
Whereas 'as TableValue' will return null and we'll get a nullref later - which is harder to diagnose.

Reply:
Removed the cast since we are working with the serialized version of it